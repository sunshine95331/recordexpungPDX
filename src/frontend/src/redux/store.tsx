// See the following guides for an explanation:
// https://redux-starter-kit.js.org/usage/usage-guide
// https://redux.js.org/recipes/usage-with-typescript
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

// Reducers:
import { searchReducer } from './search/reducer';

const rootReducer = combineReducers({
  search: searchReducer,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware()],
});

export type AppState = ReturnType<typeof rootReducer>;

export default store;
