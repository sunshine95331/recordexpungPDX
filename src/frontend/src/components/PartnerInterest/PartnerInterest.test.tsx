import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import store from "../../redux/store";
import PartnerInterest from ".";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <PartnerInterest email="" invalidEmail={false} />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
