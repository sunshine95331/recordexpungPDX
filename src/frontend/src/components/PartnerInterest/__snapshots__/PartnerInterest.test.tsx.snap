// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="f6 f5-ns f4-l navy bg-lightest-blue1"
>
  <main
    className="mw7 center ph4 ph5-ns pt5 pb6"
  >
    <section
      className="lh-copy mb5"
    >
      <h1
        className="mw6-l f3 f2-l fw9 lh-solid pr4-l ma0 mb4"
      >
        Provide expungement help with RecordSponge
      </h1>
      <p
        className="mb4"
      >
        RecordSponge is made for organizations to become expungement service providers. We provide both the software and supervision by volunteer attorneys â€“ organizations provide the clients and, to a large extent, the expungement service.
      </p>
      <p
        className="mb4"
      >
        We are therefore looking to partner with organizations who have contact with many people with criminal records. If you would like to learn more about partnering please get in contact below.
      </p>
      <p
        className="mb2"
      >
        You will need an Oregon eCourt Case Information (OECI) account to search for criminal records with RecordSponge, otherwise there is no additional charge.
      </p>
      <p
        className="mb4"
      >
        <a
          className="link hover-dark-blue bb"
          href="https://www.courts.oregon.gov/services/online/Pages/ojcin-signup.aspx"
        >
          You can purchase a subscription here
        </a>
        .
      </p>
      <p
        className="mb2"
      >
        No OECI account yet? The demo version has all the same features besides the ability to search the OECI database. There are examples provided or you can even enter records manually.
      </p>
      <p
        className="mb4"
      >
        <a
          className="fw7 dark-blue link hover-navy nowrap"
          href="/demo-record-search"
          onClick={[Function]}
        >
          Check out the demo version
          <span
            className="fas fa-arrow-right pl1"
          />
        </a>
      </p>
    </section>
    <section>
      <div
        className=""
        id="mc_embed_signup"
      >
         
        <form
          action="https://recordsponge.us10.list-manage.com/subscribe/post?u=8aa8348c6b5b43cde29949c59&id=17b2f23a63"
          className=""
          id="mc-embedded-subscribe-form"
          method="post"
          name="mc-embedded-subscribe-form"
          noValidate={true}
          target="_blank"
        >
          <div
            className="bg-white shadow br3 pb5 mb3"
            id="mc_embed_signup_scroll"
          >
            <span
              className="db center bb bw3 b--blue mb4"
            />
            <h2
              className="visually-hidden"
            >
              Subscribe
            </h2>
            <p
              className="fw7 ph4 ph5-ns mh3-l mb3"
            >
              We ask anyone using the software to be in touch so that we can better maintain, scale, and improve our work and community.
            </p>
            <p
              className="ph4 ph5-ns mh3-l mb4 mb5-l"
            >
              Please complete this contact form even if you already have an OECI account.
            </p>
            <div
              className="ph4 ph5-ns ph6-l"
            >
              <div
                className="mb3 mc-field-group"
              >
                <label
                  className="db fw6 mb1"
                  htmlFor="mce-EMAIL"
                >
                  Email Address (required)
                </label>
                <input
                  className="w-100 b--black-20 br2 pa3 required email"
                  id="mce-EMAIL"
                  name="EMAIL"
                  onChange={[Function]}
                  type="email"
                />
              </div>
              <div
                className="mb3 mc-field-group"
              >
                <label
                  className="db fw6 mb1"
                  htmlFor="mce-NAME"
                >
                  Name
                   
                </label>
                <input
                  className="w-100 b--black-20 br2 pa3"
                  id="mce-NAME"
                  name="NAME"
                  type="text"
                />
              </div>
              <div
                className="mb2 mc-field-group"
              >
                <label
                  className="db fw6 mb1"
                  htmlFor="mce-ORG"
                >
                  Organization
                   
                </label>
                <input
                  className="w-100 b--black-20 br2 pa3"
                  id="mce-ORG"
                  name="ORG"
                  type="text"
                />
              </div>
              <div
                className="checkbox mb4 mc-field-group input-group"
              >
                <input
                  id="mce-group[19029]-19029-0"
                  name="group[19029][1]"
                  type="checkbox"
                  value="1"
                />
                <label
                  className="fw6"
                  htmlFor="mce-group[19029]-19029-0"
                >
                  I'm interested in a demonstration of the software!
                </label>
              </div>
              <div
                className="clear"
                id="mce-responses"
              >
                <div
                  className="response"
                  id="mce-error-response"
                  visually-hidden="true"
                />
                <div
                  className="response"
                  id="mce-success-response"
                  visually-hidden="true"
                />
              </div>
               
              <div
                className="clear"
              >
                <input
                  className="w-100 pointer bg-blue white bg-animate hover-bg-dark-blue bn fw6 br2 pv3 ph4"
                  id="mc-embedded-subscribe"
                  name="subscribe"
                  onClick={[Function]}
                  type="submit"
                  value="Subscribe"
                />
              </div>
              <div
                className="black-70 w-100"
                role="alert"
              />
            </div>
          </div>
        </form>
      </div>
    </section>
  </main>
</div>
`;
